### ISSUE TITLE
[question] Are empty string and False option values equal for Conan?

### ISSUE THREAD

**Issue by db4** [conan-io/conan#10803]
- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

I have a recipe with an option specified as
```
    options = {
        'an_option': [True, False, ''],
        ...
    }
```
It turned out that `False` and `''` values are equal for Conan: they generate the same package ID. Is it intentional behavior?

**Comment by memsharded** [comment]
Yes, options are evaluated as "falsey" with "off", "0", "false", "", "None" and they are excluded from package_id. It was a mechanism to allow introducing new options without breaking existing package_ids.

I think Conan 2.0 will make it more strict, and only exclude the ``None`` one.

**Comment by db4** [comment]
> I think Conan 2.0 will make it more strict, and only exclude the `None` one.

Yes, please do it!
